jobserver/github.py

```diff
@@ -114,6 +116,18 @@ def _get_page(session, org, cursor):
 
     r = session.post("https://api.github.com/graphql", json=payload)
     r.raise_for_status()
+    result = r.json()
+
+    # In some cases graphql will return a 200 response when there are errors.
+    # https://sachee.medium.com/200-ok-error-handling-in-graphql-7ec869aec9bc
+    # Handling things robustly is complex and query specific, so here we simply
+    # take the absence of 'data' as an error, rather than the presence of
+    # 'errors' key.
+    if "data" not in result:
+        raise RuntimeError(
+            f"graphql query failed\nquery: {query}\nresponse: {json.dumps(result, indent=2)}"
```

Comment:
Very minor, but given that both `query` and `result` will have newlines it might give more readable errors to separate a bit more e.g.
```py
f"graphql query failed\n\nquery:\n{query}\n\nresponse:\n{json.dumps(result, indent=2)}"
```